lib/elixir/lib/module/types/descr.ex

```diff
@@ -2383,6 +2467,77 @@ defmodule Module.Types.Descr do
     :maps.iterator(open) |> :maps.next() |> map_literal_intersection_loop(closed)
   end
 
+  # At least one tag is a tag-domain pair.
+  defp map_literal_intersection(tag1, map1, tag2, map2) do
+    # For a closed map with domains intersected with an open map with domains:
+    # 1. The result is closed (more restrictive)
+    # 2. We need to check each domain in the open map against the closed map
+    default1 = tag_to_type(tag1)
+    default2 = tag_to_type(tag2)
```

Comment:
Is this correct? I am thinking we don't want tag_to_type but rather the value of the atom domain key, if one?

Reply:
Nevermind, this is exactly what the code does. :D